### PR TITLE
Add leafcast extension

### PR DESCRIPTION
**Pull request by syropian**
## Description

Control your Nanoleaf panels directly in Raycast!

## Screencast

https://github.com/raycast/extensions/assets/749061/8f2d5a5d-e5b0-4ab1-8f42-3f2142452d4f

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/leafcast/src/components/SetDeviceAddressForm.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+  async function attemptAutoScan() {
+    networkScan.revalidate();
+  }
+
+  return (
+    <Form
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm onSubmit={handleSetDeviceAddress} title="Save Device Address" />
```
Comment (pernielsentikaer) [inline extensions/leafcast/src/components/SetDeviceAddressForm.tsx:64]: Missing an icon

File: extensions/leafcast/src/pair-device.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  function resetDeviceAddress() {
+    setDeviceAddress("");
+  }
+
+  return deviceAddress ? (
+    <List
+      actions={
+        <ActionPanel>
+          <Action title="Pair Device" onAction={attemptToPairDevice} />
+          <Action title="Reset Device Address" style={Action.Style.Destructive} onAction={resetDeviceAddress} />
```
Comment (pernielsentikaer) [inline extensions/leafcast/src/pair-device.tsx:81]: Missing icons

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution, and happy Movember 🧔‍♂️ 💪 

I have now tested your extension, and I have some feedback ready for you:

- Could you add a [metadata image](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by syropian** [comment]
@pernielsentikaer Thanks for the review! I tried using Raycast's screenshot tool to generate my metadata assets but it seemed to only output the image at half the required dimensions. Is this a known issue? It's causing the linter to fail 😢

**Comment by pernielsentikaer** [comment]
Hi @syropian 👋 

You might need to take it on our retina display on the mac, could you try that? Otherwise I'm happy to help :)

**Pull request merged**